Add documentation on import delete profiles CSV (#19)
chamaeleonidae/api: docs/apis/imports.md

}
```

> ##### Using `kind=delete_csv` to tag User Profiles via a User ID :id=examples-profiles-delete-by-id

Reviewer: ```suggestion
##### Using `kind=delete_csv` to delete User Profiles via a User ID :id=examples-profiles-delete-by-id
```

> This will be the same User ID you send to Chameleon when calling `chmln.identify` => [Identifying Users](js/profiles.md).

With a CSV `file` like this, specify the header of `User ID` as mapping to the Chameleon User Profile field of `uid`.

```text
User ID
5a1fe53
621f8e7
```

Request:

```json
{
  "kind": "delete_csv",

>   "name": "Feedback Request: Post-BETA1",

Reviewer: ```suggestion
  "name": "Data deletion request 54Dw",
```

  "model_kind": "profile",
  "properties": [
    {
      "name": "User ID",
      "prop": "uid"
    }
  ],
  ...
}
```

Response:

```json
{
  "import": {
    ...
    "head_columns":[
      {
        "name": "User ID",
        "values": ["5a1fe53", "621f8e7"]
      }
    ]
  }
}
```

##### Using `kind=delete_csv` to delete User Profiles with Email :id=examples-profiles-delete-by-email

With a CSV `file` like this, specify the header of `Email address` as mapping to the Chameleon User Profile field of `email`.

```text
Email address
jill@sample.com
aaron@example.com
```

Request:

```json
{
  "kind": "delete_csv",

>   "name": "Feedback Request: Post-BETA2",

Reviewer: ```suggestion
  "name": "Data deletion request 65fx",
```

</details>

<details>
<summary>Tagging User Profiles using Filters</summary>

- `kind=tag_filters` + `model_kind=profile` means Tag Users Profiles matching the given Filters.

```bash
curl -X POST -H 'X-Account-Secret: ACCOUNT_SECRET' \
  -H 'Content-Type: application/json' \

>   -d '{"name": "Gmail users", "model_kind": "profile", "kind": "tag_filters", "on_model_missing": "create", "filters": [{"kind": "property", "prop": "email", "op": "in", "value": "gmail.com"}], "import_at": "$now"}' \

Reviewer: ```suggestion
  -d '{"name": "Gmail users", "model_kind": "profile", "kind": "tag_filters", "filters": [{"kind": "property", "prop": "email", "op": "in", "value": "gmail.com"}], "import_at": "$now"}' \
```

  'https://api.trychameleon.com/edit/v3/imports'
```

Optional: Check on the Import status:

```bash
curl -H 'X-Account-Secret: ACCOUNT_SECRET' 'https://api.trychameleon.com/edit/v3/imports/IMPORT_ID'
```

</details>

<details>
<summary>Deleting User Profiles by <b>UID</b></summary>

- `kind=delete_csv` + `model_kind=profile` means Delete Users Profiles by CSV.

> With a CSV like this (`feedback-request-post-BETA1.csv`):

Reviewer: ```suggestion
With a CSV like this (`data-deletion-request-54Dw.csv`):
```

```text
User ID
c4235f3
2de2c71
632665
```

First, Create the import, naming it and mapping the `User ID` CSV header to the `uid` Chameleon property:

```bash
curl -X POST -H 'X-Account-Secret: ACCOUNT_SECRET' \
  -H 'Content-Type: application/json' \

>   -d '{"name": "Feedback Request: Post-BETA1", "model_kind": "profile", "kind": "delete_csv", "on_model_missing": "create", "properties": [{"name":"User ID","prop":"uid"}] }' \

Reviewer: ```suggestion
  -d '{"name": "Data deletion request 54Dw", "model_kind": "profile", "kind": "delete_csv", "properties": [{"name":"User ID","prop":"uid"}] }' \
```

  'https://api.trychameleon.com/edit/v3/imports'
```

> Then Upload the CSV called `feedback-request-post-BETA1.csv` and trigger the import with `import_at=now`

Reviewer: ```suggestion
Then Upload the CSV called `data-deletion-request-54Dw.csv` and trigger the import with `import_at=now`
```

- Use the `import.id` from the last request in place of IMPORT_ID:

```bash
curl -X PATCH -H 'X-Account-Secret: ACCOUNT_SECRET' \

>   -F file=@feedback-request-post-BETA1.csv \

Reviewer: ```suggestion
  -F file=@data-deletion-request-54Dw.csv \
```

  'https://api.trychameleon.com/edit/v3/imports/IMPORT_ID?import_at=now'
```

Optional: Check on the Import status:

```bash
curl -H 'X-Account-Secret: ACCOUNT_SECRET' 'https://api.trychameleon.com/edit/v3/imports/IMPORT_ID'
```

</details>

<details>
<summary>Deleting User Profiles by <b>Email</b></summary>

- `kind=delete_csv` + `model_kind=profile` means Delete Users Profiles by CSV.

> With a CSV like this (`feedback-request-post-BETA1.csv`):

Reviewer: ```suggestion
With a CSV like this (`data-deletion-request-54Dw.csv`):
```

```text
Email address
jill@example.co
jess@product.io
jamie@example.com
```

First, Create the import, naming it and mapping the `Email address` CSV header to the `email` Chameleon property:

```bash
curl -X POST -H 'X-Account-Secret: ACCOUNT_SECRET' \
  -H 'Content-Type: application/json' \
  -d '{"name": "Feedback Request: Post-BETA1", "model_kind": "profile", "kind": "delete_csv", "properties": [{"name":"Email address","prop":"email"}] }' \
  'https://api.trychameleon.com/edit/v3/imports'
```

> Then Upload the CSV called `feedback-request-post-BETA1.csv` and trigger the import with `import_at=now`

Reviewer: ```suggestion
Then Upload the CSV called `data-deletion-request-54Dw.csv` and trigger the import with `import_at=now`
```

- Use the `import.id` from the last request in place of IMPORT_ID:

```bash
curl -X PATCH -H 'X-Account-Secret: ACCOUNT_SECRET' \

>   -F file=@feedback-request-post-BETA1.csv \

Reviewer: ```suggestion
  -F file=@data-deletion-request-54Dw.csv \
```

  'https://api.trychameleon.com/edit/v3/imports/IMPORT_ID?import_at=now'
```

Optional: Check on the Import status:

```bash
curl -H 'X-Account-Secret: ACCOUNT_SECRET' 'https://api.trychameleon.com/edit/v3/imports/IMPORT_ID'
```

</details>

<details>
<summary>Deleting User Profiles using Filters</summary>

- `kind=delete_filters` + `model_kind=profile` means Delete Users Profiles matching the given Filters.

```bash
curl -X POST -H 'X-Account-Secret: ACCOUNT_SECRET' \
  -H 'Content-Type: application/json' \

>   -d '{"name": "Remove users not seen in a year", "model_kind": "profile", "kind": "tag_filters", "filters": [{"kind": "property","prop":"last_seen_at","op":"lt-d","value":"365"}], "import_at": "$now"}' \

Author: ```suggestion
  -d '{"name": "Remove users not seen in a year", "model_kind": "profile", "kind": "delete_filters", "filters": [{"kind": "property","prop":"last_seen_at","op":"lt-d","value":"365"}], "import_at": "$now"}' \
```

>   -d '{"name": "Remove users not seen in a year", "model_kind": "profile", "kind": "tag_filters", "on_model_missing": "create", "filters": [{"kind": "property","prop":"last_seen_at","op":"lt-d","value":"365"}], "import_at": "$now"}' \

Reviewer: ```suggestion
  -d '{"name": "Remove users not seen in a year", "model_kind": "profile", "kind": "tag_filters", "filters": [{"kind": "property","prop":"last_seen_at","op":"lt-d","value":"365"}], "import_at": "$now"}' \
```
